oeps/best-practices/oep-0067/decisions/frontend/0011-caret-ranges-dependencies.rst

```diff
@@ -0,0 +1,55 @@
+Use Caret Ranges for npm Dependency Versions
+############################################
+
+Status
+******
+
+Proposed (February 2025)
+
+Context
+*******
+
+Managing dependencies in JavaScript projects requires balancing stability, maintainability, and performance. Historically, most projects
+have opted to pin exact dependency versions to ensure consistency. However, this approach has notable downsides, particularly when
+considering transitive dependencies, package duplication, and update workflows.
+
+Decision
+********
+
+JavaScript projects in the Open edX community should define dependencies using caret ranges (e.g., ``^1.2.3``) instead of pinning exact
+versions (e.g., ``1.2.3``).
+
+Why?
+
+#. **Pinned Dependencies Do Not Fully Lock Transitive Dependencies**: Even if a top-level dependency is pinned, its transitive dependencies
+   are not necessarily locked. This means updates can still introduce changes, making pinning less effective than intended.
+
+#. **Reduces Webpack Bundle Size and Dependency Fragmentation**: When different packages pin slightly different versions of the same dependency,
+   multiple versions of that dependency may be included in the final bundle. Using caret ranges allows different packages to share a single version,
+   reducing duplication and improving performance.
+
+#. **More Clearly Communicates Engineering Intent**: Specifying caret ranges explicitly indicates that a project intends to accept non-breaking
+   updates in accordance with semantic versioning. This improves clarity for maintainers and contributors.
+
+#. **Improves Renovate Workflow and Reduces Maintenance Overhead**: With caret ranges, Renovate still detects and opens PRs for minor and patch
```

Comment:
I think it'd be nice to link to https://docs.openedx.org/projects/openedx-proposals/en/latest/best-practices/oep-0067/decisions/frontend/0009-renovate.html here. Maybe also a quick sentence mentioning that we use renovate to automate dependency updates.

Reply:
Updated!